clang/lib/CIR/CodeGen/CIRGenTypes.cpp

```diff
@@ -72,9 +63,15 @@ mlir::Type CIRGenTypes::convertType(QualType type) {
   }
   case Type::BitInt: {
     const auto *bitIntTy = cast<BitIntType>(type);
-    resultType =
-        cir::IntType::get(cgm.getBuilder().getContext(), bitIntTy->getNumBits(),
-                          bitIntTy->isSigned());
+    if (bitIntTy->getNumBits() > cir::IntType::maxBitwidth()) {
+      cgm.errorNYI(SourceLocation(), "large _BitInt type", type);
+      resultType = cir::IntType::get(cgm.getBuilder().getContext(), 32,
```

Comment:
I might suggest we recover by doing IntType::maxBitwidth.  WDYT?

Reply:
Compilation has failed at this point, so it doesn't really matter exactly what type is used.  I like the consistency of always returning `int` for not-yet-implemented types.  (That code will become simpler when aliases for the common CIR types are created.)

Reply:
That seems like a reasonable take.  That seems consistent.